### ISSUE TITLE
mode=IP_V4_ONLY doesn't apply to getaddrinfo calls

### ISSUE THREAD

**Issue by Gillingham** [cannatag/ldap3#750]
OS: Solaris 10 sparc, Python 3.6.9, ldap3 2.6.1,

Sample test client code:
```python
import logging
logging.basicConfig(level=logging.DEBUG)

import socket
logging.info('socket.has_ipv6: %s' % socket.has_ipv6)

LDAP_SERVER = 'ldap.foo'

from ldap3 import Connection, Server, IP_V4_ONLY, IP_V4_PREFERRED
from ldap3.utils.log import set_library_log_detail_level, OFF, BASIC, NETWORK, EXTENDED

set_library_log_detail_level(EXTENDED)

ldap_server = Server(host=LDAP_SERVER, mode=IP_V4_ONLY)
#ldap_server = Server(host=LDAP_SERVER, mode=IP_V4_PREFERRED)
#ldap_server = Server(host=LDAP_SERVER)
print(ldap_server)

conn = Connection(server=ldap_server, auto_bind=True)
print(conn)
```

Output:
```
INFO:root:socket.has_ipv6: False
DEBUG:ldap3:ERROR:detail level set to EXTENDED
DEBUG:ldap3:BASIC:instantiated Server: <Server(host='ldap.foo', port=389, use_ssl=False, allowed_referral_hosts=[('*', True)], get_info='SCHEMA', mode='IP_V4_ONLY')>
ldap://ldap.foo:389 - cleartext
DEBUG:ldap3:BASIC:instantiated <SyncStrategy>: <ldap://ldap.foo:389 - cleartext - user: None - not lazy - unbound - closed - <no socket> - tls not started - not listening - No strategy - internal decoder - async - real DSA - not pooled - cannot stream output>
DEBUG:ldap3:BASIC:performing automatic bind for <ldap://ldap.foo:389 - cleartext - user: None - not lazy - unbound - closed - <no socket> - tls not started - not listening - SyncStrategy - internal decoder>
DEBUG:ldap3:NETWORK:opening connection for <ldap://ldap.foo:389 - cleartext - user: None - not lazy - unbound - closed - <no socket> - tls not started - not listening - SyncStrategy - internal decoder>
DEBUG:ldap3:BASIC:address for <ldap://ldap.foo:389 - cleartext> resolved as <[<AddressFamily.AF_INET6: 26>, <SocketKind.SOCK_STREAM: 2>, 6, '', (26, b'\x01\x85\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')]>
DEBUG:ldap3:ERROR:invalid server address for <ldap://ldap.foo:389 - cleartext - user: None - not lazy - unbound - closed - <no socket> - tls not started - not listening - SyncStrategy - internal decoder>
Traceback (most recent call last):
  File "test-ldap.py", line 19, in <module>
    conn = Connection(server=ldap_server, auto_bind=True)
  File "/tps/lib/python3.6/site-packages/ldap3/core/connection.py", line 326, in __init__
    self.do_auto_bind()
  File "/tps/lib/python3.6/site-packages/ldap3/core/connection.py", line 341, in do_auto_bind
    self.open(read_server_info=False)
  File "/tps/lib/python3.6/site-packages/ldap3/strategy/sync.py", line 56, in open
    BaseStrategy.open(self, reset_usage, read_server_info)
  File "/tps/lib/python3.6/site-packages/ldap3/strategy/base.py", line 150, in open
    raise LDAPSocketOpenError('invalid server address')
ldap3.core.exceptions.LDAPSocketOpenError: invalid server address
```

I assume what should happen, is if mode is set to IP_V4_ONLY than the socket.getaddrinfo calls should also force IPV4 lookups. The extra `socket.AI_ADDRCONFIG | socket.AI_V4MAPPED` used there now seem to not do that, causing this bug for me.

**Comment by cannatag** [comment]
Hi, can you check the code in dev? It should work now.

Thanks,
Giovanni

**Comment by Gillingham** [comment]
The posted test code now works as expected, tested on commit c6bd565087b64689009fb34105dfdf217b53aa53

**Comment by cannatag** [comment]
Thanks. The fix will be included in v2.6.2.

**Comment by cannatag** [comment]
Fixed in ldap3 v2.7.
